d3a_api_client/commands.py

```diff
@@ -0,0 +1,72 @@
+from d3a_api_client.enums import Commands, command_enum_to_command_name
+
+
+class ClientCommandBuffer:
+
+    def __init__(self, ):
+        self._commands_buffer = []
+
+    def offer_energy(self, area_uuid, energy, price):
+        area_uuid = area_uuid
+        args = {energy, price}
+        action = Commands.OFFER
+        self.execute(area_uuid, action, args)
+
+    def offer_energy_rate(self, area_uuid, energy, rate):
+        area_uuid = area_uuid
+        args = {energy, rate}
```

Comment:
IIRC these arguments should not be a set but a dict. You need to construct the argument dict, for instance https://github.com/gridsingularity/d3a-api-client/blob/master/d3a_api_client/rest_device.py#L87